pkg/query-service/postprocess/table.go

```diff
@@ -0,0 +1,299 @@
+package postprocess
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"go.signoz.io/signoz/pkg/query-service/constants"
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+)
+
+func getAutoColNameForQuery(queryName string, params *v3.QueryRangeParamsV3) string {
+	q := params.CompositeQuery.BuilderQueries[queryName]
+	if q.DataSource == v3.DataSourceTraces || q.DataSource == v3.DataSourceLogs {
+		if q.AggregateAttribute.Key != "" {
+			return fmt.Sprintf("%s(%s)", q.AggregateOperator, q.AggregateAttribute.Key)
+		}
+		return string(q.AggregateOperator)
+	} else if q.DataSource == v3.DataSourceMetrics {
+		if q.SpaceAggregation != "" && params.Version == "v4" {
+			return fmt.Sprintf("%s(%s)", q.SpaceAggregation, q.AggregateAttribute.Key)
+		}
+		return fmt.Sprintf("%s(%s)", q.AggregateOperator, q.AggregateAttribute.Key)
+	}
+	return queryName
+}
+
+func TransformToTableForBuilderQueries(results []*v3.Result, params *v3.QueryRangeParamsV3) []*v3.Result {
+	if len(results) == 0 {
+		return []*v3.Result{}
+	}
+
+	// Sort results by QueryName
+	sort.Slice(results, func(i, j int) bool {
+		return results[i].QueryName < results[j].QueryName
+	})
+
+	// Create a map to store all unique labels
+	seen := make(map[string]struct{})
```

Comment:
why not `map[string]bool`

Reply:
https://medium.easyread.co/golang-series-empty-struct-ed317e6d8600

Reply:
nice read 👍 , can we add the same as an inline comment or the reason for doing the same inline as might be confusing for someone else seeing this later